api/compile.js

```diff
@@ -65,30 +68,34 @@ module.exports.compileLinksFile = function (req, res, next) {
   // console.log('compile socket connected')
     socket.on('compile', function () {
       // console.log(`Compiling Tutorial ${tutorialId} for user ${username}`)
-      killLinksProc()
       var promises = [module.exports.createConfigFile(username),
         module.exports.createSourceFile(username, tutorialId)]
       Promise.all(promises)
         .then(() => {
-          module.exports.linxProc = spawn('linx', [`--config=tmp/${username}_config`, `tmp/${username}_source.links`])
-          module.exports.linxProc.stdout.on('data', (data) => {
+          killLinksProc(username)
+          let linxProc = spawn('linx', [`--config=tmp/${username}_config`, `tmp/${username}_source.links`])
+          linxProc.stdout.on('data', (data) => {
             socket.emit('compile error', 'STDOUT: ' + data.toString())
             // console.log('sent stdout: ' + data)
           })
 
-          module.exports.linxProc.stderr.on('data', (data) => {
+          linxProc.stderr.on('data', (data) => {
             socket.emit('compile error', 'STDERR: ' + data.toString())
             // console.log('sent stderr: ' + data)
           })
-          sleep(2000)
+
+          module.exports.sessionMap.set(username, linxProc)
+          sleep(2500)
```

Comment:
This is my fault when first coding this but the this is incrediably confusing. Would be better to have a comment that explains that it is to wait for the compiler to finish work. Also, `2500` should be an environment variable.